includes/subscriptions/class-wc-payments-invoice-service.php

```diff
@@ -202,7 +197,7 @@ public function maybe_record_invoice_payment( int $order_id ) {
 		foreach ( wcs_get_subscriptions_for_order( $order, [ 'order_type' => [ 'parent', 'renewal' ] ] ) as $subscription ) {
 			$invoice_id = self::get_subscription_invoice_id( $subscription );
 
-			if ( ! $invoice_id ) {
+			if ( ! $invoice_id || ! (bool) WC_Payments_Subscription_Service::get_wcpay_subscription_id( $subscription ) ) {
```

Comment:
```suggestion
			if ( ! $invoice_id || ! WC_Payments_Subscription_Service::is_wcpay_subscription( $subscription ) ) {
```

Can we use this function here instead? The difference is `get_wcpay_subscription_id()` just returns the Stripe subscription ID in meta, `is_wcpay_subscription()` checks the payment method and duplicate site as well. 

I don't know if we need to the weaker condition here.